include/cupla/datatypes/dim3.hpp

```diff
@@ -63,6 +63,19 @@ inline namespace CUPLA_ACCELERATOR_NAMESPACE
           t.z = z;
           return t;
         }
+
+        ALPAKA_FN_HOST_ACC
+        operator ::alpaka::vec::Vec<
+            cupla::AlpakaDim< 3u >,
+            IdxType
+        >(void) const
+        {
+            ::alpaka::vec::Vec<
+                cupla::AlpakaDim< 3u >,
+                IdxType
+            > vec(z, y, x);
```

Comment:
Is the `z, y, x` order here correct? If so, I guess this needs explanation cuz looks unexpected.

Reply:
It is correct. In alpaka the fast-moving index is `dimension - 1` in cuda it is the component `0`.

Reply:
I see. But I find it highly illogical in this class, since it's just a vector and not a kernel launch parameter (yet).

Reply:
I mean, one could theoretically use dim3 as just a 3-dimensional vector, we (and CUDA) do not forbid it. Then such a conversion would be unexpected. Perhaps it's more logical if the conversion happens somewhere inside the kernel call.

Reply:
Valid point, let us handle this behavior separately, not sure if it is possible to solve.

Reply:
As far as I understand the conversion is called only [here](https://github.com/ComputationalRadiationPhysics/cupla/blob/41d15edb6b236485893c0a3f4f60c189ee905383/include/cupla/kernel.hpp#L125-L127). In this case it might be enough to make this operator be a separate function and call it there instead of the static cast. However, that's a conclusion I made from looking at the code, did not try it.